### PR TITLE
Improve dependency specification for Sync SDK

### PR DESCRIPTION
**Pull request by berkus**
As suggested by customer.

### CODE REVIEW

File: sync-quickstart-android/build.gradle [hunk 1]
```diff
@@ -65,7 +65,9 @@ configurations {
 
 dependencies {
     implementation "androidx.appcompat:appcompat:$androidXVersion"
-    implementation 'com.twilio:sync-android-with-symbols:1.0.0'
+    implementation 'com.twilio:sync-android-with-symbols:1.0.+'
```
Comment (rusmonster) [inline sync-quickstart-android/build.gradle:68]: On android this is considered as BAD practice. At least by google. We shouldn't go that way
https://developer.android.com/studio/build/dependencies

```
Caution: When specifying dependencies, you should not use dynamic version numbers, such 
as 'com.android.tools.build:gradle:3.+'. Using this feature can cause unexpected version updates and 
difficulty resolving version differences.
```

File: sync-quickstart-android/build.gradle [hunk 2]
```diff
@@ -65,7 +65,9 @@ configurations {
...
+    // or without symbols:
```
Comment (rusmonster) [inline sync-quickstart-android/build.gradle:69]: We should add the same into the conversation demo app as well

**Review by rusmonster — CHANGES_REQUESTED** [review]

File: sync-quickstart-android/build.gradle (hunk 1 above)
Comment (berkus) [reply sync-quickstart-android/build.gradle:68]: Ok, but this does not  take semver into account, I believe. And technically should be perfectly fine for the SDK we control, as we know that patch version updates will be only bugfixes. (I.e. I definitely wouldn't do it for 3rd-party dependencies)

File: sync-quickstart-android/build.gradle (hunk 2 above)
Comment (berkus) [reply sync-quickstart-android/build.gradle:69]: Sure, go ahead.

File: sync-quickstart-android/build.gradle (hunk 1 above)
Comment (rusmonster) [reply sync-quickstart-android/build.gradle:68]: One moment we can make a mistake as any other developers. If we broke something two days before someone's "big release" noone would be happy. Even without "big release" our customers could spend days to find what's happened with sync in fresh build after they only changed font-size in a view or something like that. There is no profit in this change at all, but it adds risks for our customers. I believe we shouldn't include it in the demo app which should show best practices and in fact will be copy-pasted to a lot of customer's projects.

Comment (berkus) [reply sync-quickstart-android/build.gradle:68]: I'm sorry, are you still talking about QUICKSTART?

Comment (berkus) [reply sync-quickstart-android/build.gradle:68]: Pls update the JIRA ticket then and forward this information to Zendesk ticket too. Thanks.

**Pull request closed without merging**